src/core/lib/iomgr/event_engine/iomgr.cc

```diff
@@ -44,19 +44,15 @@ using ::grpc_event_engine::experimental::DefaultEventEngineFactory;
 using ::grpc_event_engine::experimental::EventEngine;
 using ::grpc_event_engine::experimental::Promise;
 
-// Note: This is a pointer to a shared_ptr, so it's trivially destructible.
-std::shared_ptr<EventEngine>* g_event_engine;
+EventEngine* g_event_engine = nullptr;
 
-void iomgr_platform_init(void) {
-  g_event_engine =
-      new std::shared_ptr<EventEngine>(DefaultEventEngineFactory());
-}
+void iomgr_platform_init(void) { GPR_ASSERT(g_event_engine != nullptr); }
```

Comment:
I think this should instantiate a new EE instance if `g_event_engine` is null.  That way, we continue to create our own unless the wrapped language has set one already.

If we don't do this, then we're going to require all wrapped languages *except* Python (including C++) to explicitly call `SetDefaultEventEngine()`, which I don't think is what we want.

Reply:
That's a good point. We don't yet have a default version to instantiate, I believe that branch is still operating on the `DefaultEventEngineFactory` concept from a few months back. I think we can change this in the PR that introduces that default EventEngine. Sound ok? If so, I'll leave a TODO here.

@nicolasnoble